src/modules/dashboard/components/Dialogs/AddExistingSafeDialog/CheckSafe.tsx

```diff
@@ -0,0 +1,235 @@
+import React, { useEffect } from 'react';
+import { FormikProps, useField } from 'formik';
+import {
+  defineMessages,
+  FormattedMessage,
+  MessageDescriptor,
+} from 'react-intl';
+import { isAddress } from 'web3-utils';
+import classnames from 'classnames';
+
+import Button from '~core/Button';
+import DialogSection from '~core/Dialog/DialogSection';
+import {
+  Select,
+  Input,
+  SelectOption,
+  InputStatus,
+  InputLabel,
+} from '~core/Fields';
+import { FETCH_ABORTED } from '~constants';
+import { SimpleMessageValues } from '~types/index';
+import Avatar from '~core/Avatar';
+
+import { FormValues, AddExistingSafeProps, StepProps } from './index';
+
+import styles from './AddExistingSafeDialogForm.css';
+
+const MSG = defineMessages({
+  subtitle: {
+    id: 'dashboard.AddExistingSafeDialog.CheckSafe.subtitle',
+    defaultMessage: 'Step 1: Check for the Safe',
+  },
+  chain: {
+    id: 'dashboard.AddExistingSafeDialog.CheckSafe.chain',
+    defaultMessage: 'Select Chain',
+  },
+  contract: {
+    id: 'dashboard.AddExistingSafeDialog.CheckSafe.contract',
+    defaultMessage: 'Add Safe contract',
+  },
+  safeLoading: {
+    id: 'dashboard.AddExistingSafeDialog.CheckSafe.safeLoading',
+    defaultMessage: 'Loading Safe details...',
+  },
+  safeCheck: {
+    id: 'dashboard.AddExistingSafeDialog.CheckSafe.safeCheck',
+    defaultMessage: `Safe found on {selectedChain}`,
+  },
+});
+
+export interface CheckSafeProps
+  extends Omit<AddExistingSafeProps, 'stepState'>,
+    StepProps {
+  safeDataState: [SafeData, React.Dispatch<React.SetStateAction<SafeData>>];
+  selectedChain: SelectOption;
+  setSelectedChain: React.Dispatch<React.SetStateAction<SelectOption>>;
+  baseURL: string;
+}
+
+export interface SafeContract {
+  address: string;
+  name: string;
+  displayName: string;
+  logoUri: string;
+  contractAbi: any;
+  trustedForDelegateCall: boolean;
+}
+
+export interface StatusText {
+  status: MessageDescriptor;
+  statusValues?: SimpleMessageValues;
+}
+
+type SafeData = SafeContract | undefined | null | {} | 'alreadyExists';
```

Comment:
You have the same type defined here: https://github.com/JoinColony/colonyDapp/pull/3858/files#diff-17516f245a322de3ed786172e8d5da66e2ee65d4a4b2d163242ffea703a56d10R22. Maybe you could share it between these two files?

Reply:
Good idea